### DISCUSSION TITLE
Help with Next step handler

### DISCUSSION THREAD

**Discussion by jyr2711** [eternnoir/pyTelegramBotAPI#1771] [Q&A]
Hi all. 

I need to make Next step handler to not really register the Next step unless the user id is the same as the first message.

In a group, if I start the command but a different message arrives in between the command and my response, the bot gets that message as a response. 

Tried to filter by if message.user_from.id != Id but It only handles the first message. So if the first message is ignored, the handler stops working and the Next message is not being registered...

Any help?

Maybe something like stop until a message from id arrives, but dont know how to do It. 

Thank you

**Comment by coder2020official** [comment]
Code. I'd recommend using states though. Sorry for long response.

**Reply by jyr2711** [reply]
Thank you . I read about how Next message handler works and I did the same but manually, with my own functions, and now It works. 😉
